stac_fastapi/pgstac/stac_fastapi/pgstac/models/links.py

```diff
@@ -0,0 +1,292 @@
+    prev: str = attr.ib(kw_only=True, default=None)
+
+    def link_next(self) -> PaginationLink:
+        """Create link for next page."""
+        if self.next is not None:
+            method = self.request.method
+            if method == "GET":
+                href = merge_params(self.url, {"token": f"next:{self.next}"})
+                link = PaginationLink(
+                    rel=Relations.next,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=href,
+                )
+                return link
+            if method == "POST":
+                body = self.request.postbody
+                body["token"] = f"next:{self.next}"
+                return PaginationLink(
+                    rel=Relations.next,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=f"{self.request.url}",
+                    body=body,
+                )
+
+    def link_prev(self) -> PaginationLink:
+        """Create link for previous page."""
+        if self.prev is not None:
+            method = self.request.method
+            if method == "GET":
+                href = merge_params(self.url, {"token": f"prev:{self.prev}"})
+                return PaginationLink(
+                    rel=Relations.previous,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=href,
+                )
+            if method == "POST":
+                body = self.request.postbody
+                body["token"] = f"prev:{self.prev}"
+                return PaginationLink(
+                    rel=Relations.previous,
+                    type=MimeTypes.json,
+                    method=method,
+                    href=f"{self.request.url}",
+                    body=body,
+                )
+
+
+@attr.s
+class CollectionLinksBase(BaseLinks):
+    """Create inferred links specific to collections."""
+
+    collection_id: str = attr.ib()
+
+    def collection_link(self, rel=Relations.collection) -> Link:
+        """Create a link to a collection."""
+        return Link(
+            rel=rel,
+            type=MimeTypes.json,
+            href=self.resolve(f"/collections/{self.collection_id}"),
+        )
+
+
+@attr.s
+class CollectionLinks(CollectionLinksBase):
+    """Create inferred links specific to collections."""
+
+    def link_self(self) -> Link:
+        """Return the self link."""
+        return self.collection_link(rel=Relations.self)
+
+    def link_parent(self) -> Link:
+        """Create the `parent` link."""
+        return Link(
+            rel=Relations.parent,
+            type=MimeTypes.json,
+            href=urljoin(self.base_url, "/"),
+        )
+
+    def link_item(self) -> Link:
+        """Create the `item` link."""
+        return Link(
+            rel=Relations.item,
+            type=MimeTypes.geojson,
+            href=self.resolve(f"/collections/{self.collection_id}/items"),
+        )
```

Comment:
This should be "items" - also incorrect in the original backend, will be making a PR to fix the sqlalchemy backend. We might want to consider pulling the Link logic out into common functionality as part of another set of work.

```suggestion
    def link_items(self) -> Link:
        """Create the `items` link."""
        return Link(
            rel="items",
            type=MimeTypes.geojson,
            href=self.resolve(f"/collections/{self.collection_id}/items"),
        )
```

Reply:
fixed in latest commit